src/components/MoneyRequestAmountInput.tsx

```diff
@@ -159,6 +159,20 @@ function MoneyRequestAmountInput(
         },
     }));
 
+    useEffect(() => {
+        if (typeof amount !== 'number') {
+            return;
+        }
+        const frontendAmount = amount ? CurrencyUtils.convertToFrontendAmount(amount).toString() : '';
+        setNewAmount(frontendAmount);
+        setSelection({
+            start: frontendAmount.length,
+            end: frontendAmount.length,
+        });
+        // we want to re-initialize the state only when the selected tab or amount changes
```

Comment:
I can't suggest for some reason, let's change this to `// we want to re-initialize the state only when the amount changes`